ivy/functional/frontends/torch/linalg.py

```diff
@@ -131,6 +131,41 @@ def inv_ex(input, *, check_errors=False, out=None):
             return inputInv, info
 
 
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"1.11.0 and below": ("float16", "bfloat16")}, "torch")
+def tensorinv(input, ind=2, *, out=None):
+    not_invertible = "Reshaped tensor is not invertible"
+    prod_cond = "Tensor shape must satisfy prod(A.shape[:ind]) == prod(A.shape[ind:])"
+    input_shape = ivy.shape(input)
+    if ind > 0:
+        shape_ind_end = input_shape[:ind]
+        shape_ind_start = input_shape[ind:]
+        prod_ind_end = 1
+        prod_ind_start = 1
+        for i in shape_ind_start:
+            prod_ind_start *= i
+        for j in shape_ind_end:
+            prod_ind_end *= j
+        if prod_ind_end == prod_ind_start:
+            inverse_shape = shape_ind_start + shape_ind_end
+            input = ivy.reshape(input, shape=(prod_ind_end, -1))
+            inverse_shape_tuple = tuple([*inverse_shape])
+            try:
+                inv_ex(input, check_errors=True)
+            except RuntimeError:
```

Comment:
I am not sure if using try except clause or assert is better here, I can swap in case of request.

Reply:
I think for the product condition and for `ind > 0` it would be more comprehensive to use an assertion.